src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -7415,35 +7391,61 @@ public void LocalScope_04()
         {
             var source =
 @"scoped s1 = default;
-ref scoped s2 = ref s1;
-scoped scoped s3 = default;
-scoped ref scoped s4 = ref s1;
-ref struct scoped { }
+ref scoped s2 = ref s1; // 1
+ref @scoped s3 = ref s1;
+scoped scoped s4 = default; // 2
+scoped ref scoped s5 = ref s1; // 3
+scoped ref @scoped s6 = ref s1; // 4
+ref struct scoped { } // 5
 ";
             var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
             comp.VerifyEmitDiagnostics(
-                // (3,1): error CS8936: Feature 'ref fields' is not available in C# 10.0. Please use language version 11.0 or greater.
-                // scoped scoped s3 = default;
-                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion10, "scoped").WithArguments("ref fields", "11.0").WithLocation(3, 1),
-                // (3,15): warning CS0219: The variable 's3' is assigned but its value is never used
-                // scoped scoped s3 = default;
-                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "s3").WithArguments("s3").WithLocation(3, 15),
+                // (2,5): error CS9061: Unexpected contextual keyword 'scoped'. Did you mean 'scoped ref' or '@scoped'?
+                // ref scoped s2 = ref s1; // 1
+                Diagnostic(ErrorCode.ERR_MisplacedScoped, "scoped").WithLocation(2, 5),
+                // (2,15): error CS1525: Invalid expression term '='
+                // ref scoped s2 = ref s1; // 1
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "=").WithArguments("=").WithLocation(2, 15),
                 // (4,1): error CS8936: Feature 'ref fields' is not available in C# 10.0. Please use language version 11.0 or greater.
-                // scoped ref scoped s4 = ref s1;
+                // scoped scoped s4 = default; // 2
                 Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion10, "scoped").WithArguments("ref fields", "11.0").WithLocation(4, 1),
-                // (5,12): warning CS8981: The type name 'scoped' only contains lower-cased ascii characters. Such names may become reserved for the language.
-                // ref struct scoped { }
-                Diagnostic(ErrorCode.WRN_LowerCaseTypeName, "scoped").WithArguments("scoped").WithLocation(5, 12));
+                // (4,15): warning CS0219: The variable 's4' is assigned but its value is never used
+                // scoped scoped s4 = default; // 2
```

Comment:
Are we expecting an error on types named `scoped` in a future PR? Presumably what's happening is that the parser isn't treating the second `scoped` as a contextual keyword here?

Reply:
I don't expect a blanket error on referencing types names `scoped`. I only expect that anytime we're looking for `scoped`, we'll prefer the keyword interpretation to the identifier interpretation.
The behavior for `scoped scoped s4` is an existing issue. Filed tracking issue: [https://github.com/dotnet/roslyn/issues/62950](https://github.com/dotnet/roslyn/issues/62950)